### PR TITLE
Add DesignerAttribute to types that had it in full framework

### PR DESCRIPTION
**Pull request by safern**
Contributes towards: https://github.com/dotnet/runtime/issues/31428

cc: @DustinCampbell

### CODE REVIEW

**Comment by safern** [comment]
Even though I'm modifying ref assemblies, I'm not adding new API.

File: src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs [hunk 1]
```diff
@@ -44,7 +45,7 @@ public sealed partial class OdbcCommand : System.Data.Common.DbCommand, System.I
         public override object? ExecuteScalar() { throw null; }
         public override void Prepare() { }
         public void ResetCommandTimeout() { }
-        object System.ICloneable.Clone() { throw null; }
+        object? System.ICloneable.Clone() { throw null; }
```
Comment (ericstj) [inline src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs:48]: These nullable changes are unrelated, are you OK committing in the same PR?

File: src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj [hunk 2]
```diff
@@ -119,7 +119,6 @@
     <Compile Include="System\ComponentModel\DataObjectMethodAttribute.cs" />
     <Compile Include="System\ComponentModel\DataObjectMethodType.cs" />
     <Compile Include="System\ComponentModel\DefaultBindingPropertyAttribute.cs" />
-    <Compile Include="System\ComponentModel\DesignerAttribute.cs" />
```
Comment (ericstj) [inline src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj:122]: I'm not seeing the delete for this file.

File: src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs (hunk 1 above)
Comment (safern) [reply src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs:48]: Those came in from running /t:GenerateReferenceSource... validated the implementation and the implementation has those nullable annotations so I thought it was fine to include them to not forget all the annotation sthat needed to be exposed, but whichever you prefer I'm fine with.

File: src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj (hunk 2 above)
Comment (safern) [reply src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj:122]: Since the diff of the file is small enough and I used git mv to do it, it is shown as a rename on the diff:
https://github.com/dotnet/runtime/pull/40425/files#diff-3c1478e97792216f3370461a1209c913

Comment (ericstj) [reply src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj:122]: For some reason github is rendering as an add: https://github.com/dotnet/runtime/pull/40425/files#diff-cfe6da5dfffa7283c15c62c680231733

Also, the file still appears in your branch:  https://github.com/safern/runtime/blob/IComponentAttributes/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/DesignerAttribute.cs

File: src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs (hunk 1 above)
Comment (ericstj) [reply src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs:48]: I'm OK with it.  FYI @roji

File: src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj (hunk 2 above)
Comment (safern) [reply src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj:122]: Interesting... I used git mv... will update.

Comment (safern) [reply src/libraries/System.ComponentModel.TypeConverter/src/System.ComponentModel.TypeConverter.csproj:122]: Done. Good catch, thanks

**Review by ericstj — APPROVED** [review]

File: src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs (hunk 1 above)
Comment (roji) [reply src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs:48]: Thanks for this @safern, I should have done this as part of the original annotation (but was unaware how to).

Should we have a build step to fail the CI for these kind of nullability discrepancies between ref and src?

Comment (safern) [reply src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs:48]: We do want to do that however our tooling to detect that (APICompat) is built on top of CCI so detecting this discrepancies is very hard because the nullable annotations can be expressed differently because they are stored in metadata so we can't do a token based comparison. We think it would be a better investment moving the tooling to use Roslyn and take advantage of the language APIs for this instead of having to implement ourselves every new language feature.

I have a prototype of this but haven't had funded time to finish that.

Comment (roji) [reply src/libraries/System.Data.Odbc/ref/System.Data.Odbc.cs:48]: Thanks for the update @safern, makes sense... FWIW I'm also suffering from the lack of APIs to expose NRT information in reflection (https://github.com/dotnet/runtime/issues/29723).

**Comment by RussKie** [comment]
Please add me to the notification list for this kind of changes, this broke a number of tests in Windows Forms repo, and tracking an unknown change through multiple repos isn't easy. Thank you.

**Comment by safern** [comment]
Of course. Sorry about that. I cc'ed @DustinCampbell -- I'll make sure to include you in further changes of this sort.
